src/packages/fetch-engine/src/getLatestTag.ts

```diff
@@ -66,41 +74,61 @@ export async function getLatestTag(): Promise<any> {
           engine,
           extension,
         )
-        promises.push(
-          queue.add(async () => ({
-            downloadUrl,
-            exists: await urlExists(downloadUrl),
-          })),
-        )
+        urls.push(downloadUrl)
       }
     }
   }
 
-  // wait for all queue items to finish
-  const exist: Array<{
-    downloadUrl: string
-    exists: boolean
-  }> = await Promise.all(promises)
-
-  const missing = exist.filter((e) => !e.exists)
-  if (missing.length > 0) {
-    throw new Error(
-      `Could not get new tag, as some assets don't exist: ${missing
-        .map((m) => m.downloadUrl)
-        .join(', ')}`,
-    )
-  }
+  return urls
+}
 
-  return commit
+async function getFirstFinishedCommit(commits: string[]): Promise<string> {
+  for (const commit of commits) {
+    const urls = getAllUrls(commit)
+    const exist = await pMap(urls, urlExists, { concurrency: 10 })
+    const hasMissing = exist.some(e => !e)
+
+    if (!hasMissing) {
+      return commit
+    } else {
+      const missing = urls.filter((_, i) => exist[i])
+      // if all are missing, we don't have to talk about it
+      // it might just be a broken commit or just still building
+      if (missing.length !== urls.length) {
+        console.log(`${chalk.blueBright('info')} The engine commit ${commit} is not yet done. We're skipping it as we're in dev. The following urls are missing:\n\n${missing.join('\n')}`)
+      }
+    }
+  }
 }
 
-async function getFirstExistingCommit(commits: string[]): Promise<string> {
+async function getCommitAndWaitIfNotDone(commits: string[]): Promise<string> {
   for (const commit of commits) {
-    const exists = await urlExists(
-      getDownloadUrl('all_commits', commit, 'darwin', 'query-engine'),
-    )
-    if (exists) {
+    const urls = getAllUrls(commit)
+    let exist = await pMap(urls, urlExists, { concurrency: 10 })
+    let hasMissing = exist.some(e => !e)
+
+    if (!hasMissing) {
       return commit
+    } else {
+      let missing = urls.filter((_, i) => exist[i])
+      // if all are missing, we don't have to talk about it
+      // it might just be a broken commit or just still building
+      if (missing.length !== urls.length) {
+        const started = Date.now()
+        while (hasMissing) {
+          if ((Date.now() - started) > 1000 * 60 * 20) {
+            throw new Error(`No new engine for commit ${commit} ready after waiting for 20 minutes.`)
+          }
+          console.log(`The engine commit ${commit} is not yet done. ${missing.length} urls are missing. Trying again in 5 seconds`)
+          exist = await pMap(urls, urlExists, { concurrency: 10 })
+          missing = urls.filter((_, i) => exist[i])
+          hasMissing = exist.some(e => !e)
+          if (!hasMissing) {
+            return commit
+          }
+          await new Promise(r => setTimeout(r, 5000))
```

Comment:
```suggestion
          console.log(`The engine commit ${commit} is not yet done. ${missing.length} urls are missing. Trying again in 30 seconds`)
          exist = await pMap(urls, urlExists, { concurrency: 10 })
          missing = urls.filter((_, i) => exist[i])
          hasMissing = exist.some(e => !e)
          if (!hasMissing) {
            return commit
          }
          await new Promise(r => setTimeout(r, 30000))
```

5 seconds is a lot of requests 😄 30 sounds more reasonable to me